modules/terraform/validate_struct.go

```diff
@@ -0,0 +1,99 @@
+package terraform
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/gruntwork-io/terratest/modules/collections"
+)
+
+type ValidationOptions struct {
+	RootDir string
+	//Sub-directories relative to the RootDir to walk recursively for all Terraform modules. For example, if your
+	// RootDir is /home/project and you want to validate all modules in home/project/terraform then pass "terraform"
+	// in TargetSubDirs
+	TargetSubDirs []string
+	//ExcludeDirs must be relative to the RootDir. For example if your RootDir is home/project and you want to exclude
+	// the path /home/project/test, pass "test" in ExcludeDirs
+	ExcludeDirs []string
+}
+
+// NewValidationOptions returns a ValidationOptions struct, with override-able sane defaults
+func NewValidationOptions(rootDir string, targetSubDirs, excludeDirs []string) (ValidationOptions, error) {
+	vo := ValidationOptions{
+		ExcludeDirs: []string{},
+	}
+
+	if rootDir == "" {
+		return vo, ValidationUndefinedRootDirErr{}
+	}
+	vo.RootDir = rootDir
+
+	// If no target sub directories are passed, default to recursively searching "modules" and "examples"
+	if len(targetSubDirs) == 0 {
+		vo.TargetSubDirs = []string{
+			"modules",
+			"examples",
+		}
+	} else {
+		vo.TargetSubDirs = targetSubDirs
+	}
+
+	if len(excludeDirs) > 0 {
+		var fullExclusionPaths []string
+		for _, excludedPath := range excludeDirs {
+			fullExclusionPaths = append(fullExclusionPaths, filepath.Join(rootDir, excludedPath))
+		}
+		vo.ExcludeDirs = fullExclusionPaths
+	}
+
+	return vo, nil
+}
+
+type ValidationUndefinedRootDirErr struct{}
+
+func (e ValidationUndefinedRootDirErr) Error() string {
+	return "RootDir must be defined in ValidationOptions passed to ValidateAllTerraformModules"
+}
+
+// readModuleAndExampleSubDirs returns a slice strings representing the filepaths for all valid Terraform modules
+// in both the "modules" directory and "examples" directories in the project root, if they exist.
+func ReadModuleAndExampleSubDirs(opts ValidationOptions) ([]string, error) {
+	var terraformModuleCandidates []string
+	// We want to run InitAndValidate on all valid subdirectories of both the modules and examples dirs
+
+	for _, dir := range opts.TargetSubDirs {
```

Comment:
Any reason not to use [glob](https://github.com/mattn/go-zglob) with `**/*.tf`?

Reply:
Seems to work well - done in c3e36ef